### PR TITLE
Add Workout creation screen

### PR DESCRIPTION
**Pull request by Siupahc**
# Added Features

- WorkoutCreationScreen.kt

# Summary:

This PR introduces the WorkoutCreationScreen, which works with both Yoga and BodyWeight Workout. You can now navigate to this screen when creating a workout, and clicking on "Create from scratch", import is currently WIP.

# Implementation Details:

I added a small enum in Workout, called **WorkoutType**, whis is simply BODY_WEIGHT, YOGA ,RUNNING, which I used to give as parameter to my WorkoutCreationScreen composable

The main idea I had was to use only one screen for both Yoga and Workout as it is pretty much the same.

However this turned out to be not really convenient, because the architectures of our data class wasn't really on point. For example I had to change the WorkoutViewModel to be covariant with the Workout. But this is all details.

I had a some issues trying to have some pop ups to show up, because navigating in the middle of the process of creating a workout is really hard to handle, so there's a lot of if else in my code, because there are many booleans to show or not show some composables.

# Future Work :

I'll make this code cleaner, it's currently a bit of a mess.
Also I'll make the UI prettier, I saw what we have done in the mainscreen, and I'll try to reuse this work.
I'll have to implement the import part, but this should be easy as I simply need to create the screen, because my code already takes into account the fact that we can edit a workout.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt [hunk 1]
```diff
@@ -26,8 +30,12 @@ object Screen {
...
   const val PREFERENCES = "Preferences Screen"
   const val SETTINGS = "Settings Screen"
+  const val BODY_WEIGHT_CREATION = "BodyWeightTraining Screen"
+  const val RUNNING_CREATION = "RunningTraining Screen"
+  const val YOGA_CREATION = "YogaTraining Screen"
   const val SESSIONSELECTION = "Session Selection Screen"
-  const val IMPORTORCREATE = "Import Or Create Screen"
+  const val IMPORTORCREATE_BODY_WEIGHT = "Import Or Create Body Weight Screen"
+  const val IMPORTORCREATE_YOGA = "Import Or Create Yoga Screen"
   const val TEST = "Test Screen"
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt:39]: I see the problem, and adding Screens and Routes to differenctiate which type is called in the classed is a bit ugly.

Maybe there's a better way of doing it?

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+@Composable
+fun WorkoutCreationScreen(
+    navigationActions: NavigationActions,
+    workoutType: WorkoutType,
+    workoutViewModel: WorkoutViewModel<Workout>,
+    isImported: Boolean
+) {
+  val context = LocalContext.current
+  val selectedWorkout =
+      if (isImported) workoutViewModel.selectedWorkout.collectAsState().value else null
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:57]: Good way to reuse this screen for the import feature

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        },
+        dismissButton = {
+          Button(
+              onClick = { showExerciseDialog = false },
+              modifier = Modifier.testTag("addExerciseCancelButton")) {
+                Text("Cancel")
+              }
+        })
+  }
+}
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:322]: As you said, this screen needs work to make it more attractive.

Try annotating your code to make it easier to read.

**Review by didi2101 — APPROVED** [review]
Good functional feature that needs refining, especially for the WorkoutCreationScreen.

I will approve this code for the M1 demo.

File: app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt (hunk 1 above)
Comment (Siupahc) [reply app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt:39]: Yes it is ugly, no I didn't find another way.

**Pull request merged**